shared/js/background/chrome-events.es6.js

```diff
@@ -201,26 +219,32 @@ chrome.webRequest.onHeadersReceived.addListener(
             return ATB.updateSetAtb(request)
         }
 
+        let responseHeaders = request.responseHeaders
+
+        if (isFlocEnabled && (request.type === 'main_frame' || request.type === 'sub_frame')) {
+            // there can be multiple permissions-policy headers, so we are good always appending one
+            responseHeaders.push({ name: 'permissions-policy', value: 'interest-cohort=()' })
+        }
+
         if (blockingExperimentActive()) {
-            let responseHeaders = request.responseHeaders
             // Strip 3rd party response header
             const tab = tabManager.get({ tabId: request.tabId })
-            if (!request.responseHeaders) return
-            if (tab && tab.site.whitelisted) return
+            if (!request.responseHeaders) return { responseHeaders: responseHeaders }
```

Comment:
```suggestion
            if (!request.responseHeaders) return { responseHeaders }
```

Reply:
This also suggests sometimes we have no headers, is this on non document like requests? Like `javascript:` perhaps?

Perhaps we have to handle that above too when using responseHeaders.push?

Reply:
Can we safely use `{ responseHeaders }`? I haven't seen that in other parts of the code.

> This also suggests sometimes we have no headers, is this on non document like requests? Like javascript: perhaps?
> Perhaps we have to handle that above too when using responseHeaders.push?

Hm, this is good point, I'll ask Brad - he added that condition.

Reply:
OK we do use the `{ x }` notation in other places 👍

Brad told me that he can't recall in which cases headers were empty, but he remember this happening. I added a safeguard.